src/main/java/mandel/projectile/ProjectileFrame.java

```diff
@@ -0,0 +1,107 @@
+package mandel.projectile;
+
+import javax.swing.*;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import javax.swing.event.DocumentEvent;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class ProjectileFrame extends JFrame
+{
+    public ProjectileFrame() {
+        setSize(400, 600);
+        setTitle("Projectile Calculator");
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        setLayout(new GridLayout(8, 2));
+
+        JLabel velocityLabel = new JLabel("Velocity");
+        JTextField velocityField = new JTextField();
+        add(velocityLabel);
+        add(velocityField);
+
+        JLabel angleLabel = new JLabel("Angle");
+        add(angleLabel);
+        JSlider angleSlider = new JSlider(0, 90);
+        angleSlider.setMajorTickSpacing(10);
+        angleSlider.setMinorTickSpacing(5);
+        angleSlider.setPaintTicks(true);
+        angleSlider.setPaintLabels(true);
+        add(angleSlider);
+
+        JLabel secondsLabel = new JLabel("Seconds");
+        JTextField secondsField = new JTextField();
+        add(secondsLabel);
+        add(secondsField);
+
+        JLabel labelX = new JLabel("X");
+        JLabel x = new JLabel();
+        add(labelX);
+        add(x);
+
+        JLabel labelY = new JLabel("Y");
+        JLabel y = new JLabel();
+        add(labelY);
+        add(y);
+
+        JLabel peakLabelY = new JLabel("PeakY");
+        JLabel peakY = new JLabel();
+        add(peakLabelY);
+        add(peakY);
+
+        JLabel interceptLabelX = new JLabel("InterceptX");
+        JLabel interceptX = new JLabel();
+        add(interceptLabelX);
+        add(interceptX);
+
+        JButton calculateButton = new JButton("Calculate");
+        add(new JLabel());
+        add(calculateButton);
+
+        calculateButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                Projectile projectile = new Projectile(angleSlider.getValue(),
+                        Double.parseDouble(velocityField.getText()));
+                projectile.setSeconds(Double.parseDouble(secondsField.getText()));
+
+                x.setText(Double.toString(projectile.getX()));
+                y.setText(Double.toString(projectile.getY()));
+                peakY.setText(Double.toString(projectile.getPeakY()));
+                interceptX.setText(Double.toString(projectile.getInterceptX()));
```

Comment:
Get rid of the duplicate code.

Reply:
I tried to make it non-repetitive but it wasn't working well. Is the best way to do it creating a new method and passing everything into it and then calling it each time?